File: backend/app/schemas/sensor.py
from datetime import datetime

from pydantic import BaseModel


class SensorOut(BaseModel):
    id: int
    name: str
    location: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ReadingOut(BaseModel):
    id: int
    sensor_id: int
    temperature: float
    timestamp: datetime
    is_anomaly: bool = False

    model_config = {"from_attributes": True}


class AlertOut(BaseModel):
    id: int
    sensor_id: int
    sensor_name: str
    temperature: float
    timestamp: datetime
    is_anomaly: bool = False

    model_config = {"from_attributes": True}

File: frontend/src/types/sensor.ts
export interface Sensor {
  id: number;
  name: string;
  location: string;
  created_at: string;
}

export interface Reading {
  id: number;
  sensor_id: number;
  temperature: number;
  timestamp: string;
  is_anomaly: boolean;
}

export interface Alert {
  id: number;
  sensor_id: number;
  sensor_name: string;
  temperature: number;
  timestamp: string;
  is_anomaly: boolean;
}
